crates/componentize/src/abi_conformance/mod.rs

```diff
@@ -189,38 +187,39 @@ pub async fn test(
 }
 
 pub(crate) fn create_store(engine: &Engine, test_config: TestConfig) -> Store<Context> {
-    create_store_with_context_and_wasi(engine, test_config, |_| {}, |b| b)
+    create_store_with_context_and_wasi(engine, test_config, |_| {}, |_| Ok(())).unwrap()
 }
 
 pub(crate) fn create_store_with_wasi(
     engine: &Engine,
     test_config: TestConfig,
-    wasi_builder: impl FnOnce(WasiCtxBuilder) -> WasiCtxBuilder,
-) -> Store<Context> {
+    wasi_builder: impl FnOnce(&mut WasiCtxBuilder) -> Result<()>,
+) -> Result<Store<Context>> {
     create_store_with_context_and_wasi(engine, test_config, |_| {}, wasi_builder)
 }
 pub(crate) fn create_store_with_context(
     engine: &Engine,
     test_config: TestConfig,
     context_builder: impl FnOnce(&mut Context),
 ) -> Store<Context> {
-    create_store_with_context_and_wasi(engine, test_config, context_builder, |b| b)
+    create_store_with_context_and_wasi(engine, test_config, context_builder, |_| Ok(())).unwrap()
 }
 
 pub(crate) fn create_store_with_context_and_wasi(
     engine: &Engine,
     test_config: TestConfig,
     context_builder: impl FnOnce(&mut Context),
-    wasi_builder: impl FnOnce(WasiCtxBuilder) -> WasiCtxBuilder,
-) -> Store<Context> {
+    wasi_builder: impl FnOnce(&mut WasiCtxBuilder) -> Result<()>,
+) -> Result<Store<Context>> {
```

Comment:
It seems we unwrap this result in the caller. Perhaps we should just keep this infallible and unwrap (or better yet `expect`) in the closures?